library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java

```diff
@@ -403,7 +406,16 @@ public Builder(Activity activity, boolean useNewStyle) {
          * @return the created ShowcaseView
          */
         public ShowcaseView build() {
-            insertShowcaseView(showcaseView, activity);
+            ViewGroup parent;
+            int index;
+            if (mParent == null) {
+                parent = ((ViewGroup) activity.getWindow().getDecorView());
```

Comment:
you could assign this to `mParent` in the constructor of Builder, and then you wouldn't have you check against null?